core/server/web/middleware/validation/profile-image.js

```diff
@@ -0,0 +1,23 @@
+var common = require('../../../lib/common'),
+    imageLib = require('../../../lib/image');
+
+module.exports = function profileImage() {
+    // we checked for a valid image file, now we need to do validations for profile image
+    return function profileImageValidation(req, res, next) {
+        return imageLib.blogIcon.getIconDimensions(req.file.path).then(function (response) {
```

Comment:
I think i see why you have used this function 🤔

We have:
- image-size utility (get size from local storage path or url)
- blog-icon utility (get size from any path)

The blog-icon utility describes to fetch image dimensions for .ico file. IMO it's worth making the image size utility a little clearer. 3 entry points. e.g. imageLib.getSizeFromStoragePath,  imageLib.getSizeFromUrl,  imageLib.getSizeFromPath.